src/boot/efi/log.c

```diff
@@ -7,16 +7,48 @@
 
 static unsigned log_count = 0;
 
+#define DEBUGCON_PREFIX u"systemd: "
+#define DEBUGCON_ACK 0xE9
+#define DEBUGCON_PORT 0x402
+
 void freeze(void) {
         for (;;)
                 BS->Stall(60 * 1000 * 1000);
 }
 
+#if defined(__i386__) || defined(__x86_64__)
+static bool log_has_debugcon(void) {
```

Comment:
is this really safe to call on physical hw? i.e. maybe the port is overloaded with something else there which we better shouldn't touch?

given we already have the in_hypervisor() helper in the EFI codebase, maybe just call this first, and return false ifit doesn't check out? better be safe than sorry?

Reply:
Yep, checking `in_hypervisor()` is a trivial addition, so might as well be paranoid.